src/main/java/de/braintags/vertx/util/json/JsonDiff.java

```diff
@@ -481,6 +483,142 @@ private static void squashArrayDiff(final ArrayNode base, final ArrayNode diff,
     base.addAll(Arrays.asList(newValues));
   }
 
+  /**
+   * Retains the diff tree specified by retainedFields
+   *
+   * @param base
+   *          is not modified
+   * @param retainedFields
+   *          is not modified
+   * @param diff
+   *          MAY BE MODIFIED!
+   * @return
+   *         the modified JsonNode
+   */
+  public static JsonNode retainDiffTree(final JsonNode base, final JsonNode retainedFields, final JsonNode diff) {
+    return internalRetainDiffTree(base, retainedFields, diff, Json.mapper.getNodeFactory(), false);
+  }
+
+  private static JsonNode internalRetainDiffTree(final JsonNode base, final JsonNode retainedFields,
+      final JsonNode diff,
+      final JsonNodeFactory nodeFactory, final boolean arrayMapsConverted) {
+    JsonNodeType nodeType = base.getNodeType();
+    if (nodeType == null) {
```

Comment:
`nodeType == null` is never happening during the test

Reply:
done